### DISCUSSION TITLE
Avalonia canvas performance compared with WPF

### DISCUSSION THREAD

**Discussion by charleypeng** [AvaloniaUI/Avalonia#12372] [General]
platform: windows10 .19045.3208

I am using canvas to do some test, but I found that when I add controls into canvas more than certain number(likne 10000), the UI becomes laggy and almost not usable,  while the same code in WPF, it comsums a lot of computer reousrces of course but the UI can still be working and can be used(yet a little bit laggy). Is it because Avalonia is using SKIA and the render is not using graphic card or just the Canvas performance is not good enough?

here's the sample code:
[Sample](https://github.com/charleypeng/Cagaya_Radar.git)

[WPF project](https://github.com/charleypeng/Cagaya_Radar/tree/main/wpf/Cagaya)

**Comment by dbriard** [comment]
May be your app require more GPU memory than the default Avalonia value.
You could add this to change the default among of memory allowed to Skia (default value is very low).
![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/bbe51351-a9f7-4d14-a26c-f0bbf20f11db)

WPF use as much memory as available on the system.
Skia on the other hand, you have to decide how much memory can be used. And if your app need more, Skia is continuously swapping.

**Reply by charleypeng** [reply]
it's working!  thanks for the tip
